libs/types/src/cdf/ballot-definition/convert.ts

```diff
@@ -1070,10 +1070,31 @@ export function convertCdfBallotDefinitionToVxfElection(
       // context).
       assert(ballotStyle.ExternalIdentifier.length === 1);
 
+      const ballotStyleId = ballotStyle.ExternalIdentifier[0].Value;
+
+      const idParts = ballotStyleId.split('_');
+
+      // Check if this is a VxDesign formatted Id_LanguageCode Ballot Id.
+      // If so extract the group ID from the first part of the Id.
+      if (
+        ballotStyle.Language &&
+        ballotStyle.Language.length === 1 &&
+        idParts.length === 2 &&
+        idParts[1] === ballotStyle.Language[0]
+      ) {
+        return {
+          id: ballotStyleId as Vxf.BallotStyleId,
+          groupId: idParts[0] as Vxf.BallotStyleGroupId,
+          districts: districtIds,
+          precincts: precinctIds,
+          partyId: ballotStyle.PartyIds?.[0] as Vxf.PartyId | undefined,
+          languages: ballotStyle.Language,
+        };
```

Comment:
nit: Since it's only the `groupId` that varies across the two cases (VxDesign-formatted ballot ID vs. not), maybe we just vary that instead of a fully separate return. Just to avoid accidentally updating one return and not the other down the road when we want to edit one of the other fields.

```ts
const groupId = isVxDesignFormattedBallotId ? idParts[0] : ballotStyleId;
```